lms/djangoapps/courseware/views/views.py

```diff
@@ -1647,14 +1647,19 @@ def render_xblock(request, usage_key_string, check_if_enrolled=True):
             raise Http404("Course not found.")
 
         # get the block, which verifies whether the user has access to the block.
+        recheck_access = request.GET.get('recheck_access') == '1'
```

Comment:
I trust that you have thought about this, but I just wanna understand this approach. This feels like a reasonable way to allow a caller to ask for this to happen. I just don't understand *why* we're rechecking access this way, vs whatever the legacy view does. Hit me up on slack, just so I can learn something.

Reply:
Does the learning MFE have to do any work before this access check code works? If not, maybe we should just always set the `recheck_access` flag, rather than making it conditional in the API?

Reply:
The MFE does not have to do any additional work with the current implementation. I kept away from always rechecking access because I didn't want to change the behavior for other callers to this method (namely mobile I believe). By rechecking access, the endpoint will go from not returning any xblocks that have access restrictions to returning them with the access denied message. Since other consumers wouldn't be expecting that, I went with making it conditional in the API.

Reply:
Makes sense.